javascripts/discourse/initializers/post-height-detective.js

```diff
@@ -14,80 +14,61 @@ export default {
           // otherwise we might miss something
-          this.checkForHeightChanges();
+          this.trackElements();
         },
 
-        checkForHeightChanges() {
+        trackElements() {
           this.element
             .querySelectorAll("article[data-post-id]")
-            .forEach(this.trackHeight);
+            .forEach((e) => this.resizeObserver.observe(e));
         },
 
-        tick() {
-          const nextTick = later(
-            this,
-            () => {
-              this.checkForHeightChanges();
-              this.tick();
-            },
-            1000
-          );
-          this.set("nextTick", nextTick);
-        },
-
-        didInsertElement() {
-          this._super(...arguments);
-          this.tick();
-        },
-
-        willDestroyElement() {
-          this._super(...arguments);
-          const nextTick = this.nextTick;
-          cancel(nextTick);
-        },
-
-        _refresh() {
-          this._super(...arguments);
-          this.recordedHeights = {};
-          this.recordedHtml = {};
+        @bind
+        resizeObserverCallback(resizeObserverEntries) {
+          for (const entry of resizeObserverEntries) {
+            this.trackHeight(entry.target, entry.contentRect.height);
+          }
         },
 
         @bind
-        trackHeight(post) {
-          const postId = post.id.split("_")[1];
+        trackHeight(post, renderedHeight) {
+          const postNumber = post.id.split("_")[1];
           const topicId = this.get("posts.posts.firstObject.topic.id");
-          const postLink = `${window.location.protocol}//${window.location.hostname}/t/${topicId}/${postId}`;
+          const postLink = `${window.location.protocol}//${window.location.hostname}/t/${topicId}/${postNumber}`;
           const postElement = post.querySelector(".cooked");
-          const renderedHeight = postElement.getBoundingClientRect().height;
-          const oldHeight = this.recordedHeights[postId];
+          const oldHeight = this.recordedHeights.get(post);
 
           if (oldHeight && renderedHeight !== oldHeight) {
             console.error(
               `🕵 Height of ${postLink} has changed after initial render. Was ${oldHeight}, now ${renderedHeight}`
             );
 
-            if (this.recordedHtml[postId] === postElement.outerHTML) {
+            const recordedHtml = this.recordedHtml.get(post);
+            if (recordedHtml === postElement.outerHTML) {
               console.log("No HTML change, check CSS");
             } else {
               console.log(
-                `Previous HTML:\n\n${this.recordedHtml[postId]}\n\nCurrent HTML:\n\n${postElement.outerHTML}`
+                `Previous HTML:\n\n${recordedHtml}\n\nCurrent HTML:\n\n${postElement.outerHTML}`
               );
             }
 
-            document
-              .querySelector("body")
-              .classList.add("suspicious-post-detected");
+            next(() =>
```

Comment:
Was this not necessary before?

Reply:
It was broken before. If you refreshed a topic page with a height-changing post, it would print a warning to the console but wouldn't light up the header red. Adding `next()` here ensures that it runs after the `onPageChange` callback.